### ISSUE TITLE
Only one person while training

### ISSUE THREAD

**Issue by skbhat** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#45]
Hi,

When I checked the code, I could see that one image had only one person annotated in the COCO dataset.  There was no mask on other portions which may contain other people.  If this understanding is correct, how does it work on multiple persons?

I am referring to this [line](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch/blob/master/datasets/coco.py#L36).  I could see keypoints available for only one person in the list.

I am asking this because, when I ran training with the COCO dataset, I am getting zero detections.  Even when I initiate my model with weights of checkpoint_iter_370000.pth (which works well), just after 5000 iterations (batch size 16), the detections on live demo becomes zero.

thanks and regards
skbhat

**Comment by Daniil-Osokin** [comment]
Hi, each label has annotations for all persons inside the image. There is one subject, around wich crop is made, and the rest, stored in `processed_other_annotations` field. For training suggest to follow the [guide](https://github.com/Daniil-Osokin/lightweight-human-pose-estimation.pytorch#training)

**Comment by skbhat** [comment]
Hi Thanks for the response.  I had read the guide, but did not notice the minor difference in training options 

--from-mobilenet  (for mobile net)
-weights-only       (for checkpoint from previous step)

**Comment by skbhat** [comment]
Hi, If one starts training from scratch, any estimation how many iteration required?

**Comment by Daniil-Osokin** [comment]
You can pick it from checkpoints name: `checkpoint_iter_420000.pth`, so it passed 420000 iterations for training from MobileNet. I believe you should start to have non-zero detections after ~5000 iterations or so.

**Comment by Daniil-Osokin** [comment]
Hope, it helped.

**Comment by rodyt** [comment]
@Daniil-Osokin To clarify, does "5000 iterations" and "420,000 iterations" mean 5000/420,000 iterations starting from the Imagenet weights (provided in Google Drive) at 80 examples per batch?

**Comment by Daniil-Osokin** [comment]
Yes. We did not do hyperparameters tuning, it might converge faster to the same numbers. One can try to run it with `ReduceLROnPlateau` scheduler.
